### DISCUSSION TITLE
Ruby?

### DISCUSSION THREAD

**Discussion by benjamin-cash** [JCSDA/spack-stack#1207] [Q&A]
Is Ruby available as part of spack-stack, and if not would it be possible to include it? rocoto requires ruby, and installing it using a version of ruby built with system libraries breaks when using spack-stack modules (or at least the containerized version).

**Comment by climbfuji** [comment]
Ruby exists as a spack package, from a first look at the dependencies it may not be too hard to add it. What's the error you are getting with ruby in the container? Something related to openssl?

**Reply by benjamin-cash** [reply]
./bin/rocotostat: /groups/BCASH/rocoto/bin/../sbin/rocotostat.rb: /opt/sw/spack/apps/linux-rhel8-x86_64_v2/gcc-10.3.0/ruby-3.1.0-4e/bin/ruby: bad interpreter: No such file or directory. 

Although in chatting with @mark-a-potts about what they did with SRW, it seems like it should be possible to run rocoto outside of the container and only run executables that need the container within it. So I might be able to just work around it.

**Reply by climbfuji** [reply]
Note. This is not a supported installation / way of installing spack-stack (inferring from the path) - container builds supported by spack-stack work differently. Is this an EPIC build/container? If so, please ask EPIC to fix this. whatever the shebang in `/opt/sw/spack/apps/linux-rhel8-x86_64_v2/gcc-10.3.0/ruby-3.1.0-4e/bin/ruby` is (if that file exists at all) must be wrong.

**Reply by benjamin-cash** [reply]
Hi @climbfuji - When I look at /opt from inside the container there is no /sw directory at all, just:
```
Singularity> ls /opt
intel  spack-stack
```
That is the path to the system install of ruby from outside of the container:
```
[bcash@hop-amd-1 rocoto]$ which ruby
/opt/sw/spack/apps/linux-rhel8-x86_64_v2/gcc-10.3.0/ruby-3.1.0-4e/bin/ruby
```

**Reply by climbfuji** [reply]
Definitely not how the spack-stack container builds look like. Someone from EPIC must have installed spack-stack in their own way in the container.

**Reply by benjamin-cash** [reply]
The containers I pulled from are here:  https://hub.docker.com/u/noaaepic. Is there another/different source for spack-stack containers? Tagging @mark-a-potts as I understand he built the noaaepic containers.
